tests: kernel: profiling: Fix _sys_soc_suspend logic

ramakrishnapallala opened the pull request:
Call k_thread_foreach only once from _sys_soc_suspend()
otherwise it will flood the console with stack dumps. As per
the test implementation, the one time call happens through
test_call_stacks_analyze_idle test case.

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

vonhust commented:
I think it's ok to only check thread->stack_info.start.  For stack size, the real size may vary with different arch.

rgundi reviewed (approved):
LGTM

nashif commented:
@ramakrishnapallala this does not fix the issue

andrewboie commented on tests/kernel/profiling/profiling_api/src/main.c:
```diff
@@ -88,9 +97,10 @@ static void thread_callback(const struct k_thread *thread, void *user_data)
-			K_THREAD_STACK_BUFFER(test_stack)) &&
-			(thread->stack_info.size == STACKSIZE)) {
+			K_THREAD_STACK_BUFFER(test_stack))
+			&& thread->stack_info.size) {
```
You could probably just drop the check of stack_info.size all together.

ramakrishnapallala replied:
Ok will do.

Commit:
tests: kernel: profiling: Fix _sys_soc_suspend logic

Call k_thread_foreach only once from _sys_soc_suspend()
otherwise it will flood the console with stack dumps. As per
the test implementation, the one time call happens through
test_call_stacks_analyze_idle test case.

Also removed the the stack size comparision logic from
thread_callback() as the actual size allocated could be
different from what is requested based on the Kconfig option
CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT

Fixes Issue #7858

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>
